packages/signers/src/web3auth/signer.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  WalletClientSigner,
+  type SignTypedDataParams,
+  type SmartAccountAuthenticator,
+} from "@alchemy/aa-core";
+import { Web3Auth } from "@web3auth/modal";
+import { createWalletClient, custom, type Hash } from "viem";
+import type {
+  Web3AuthAuthenticationParams,
+  Web3AuthUserInfo,
+} from "./types.js";
+
+export class Web3AuthSigner
+  implements
+    SmartAccountAuthenticator<
+      Web3AuthAuthenticationParams,
+      Web3AuthUserInfo,
+      Web3Auth
+    >
+{
+  inner: Web3Auth;
+  private signer: WalletClientSigner | undefined;
+
+  constructor({ inner }: { inner: Web3Auth }) {
```

Comment:
I think we can continue to improve the devx, by making this takes the params for the SDK instead.

if we want to support someone passing a web3auth instance we can make that an option as well. Maybe this should be like `Web3AuthSDKParams | { inner: Web3Auth}` or we can add a static method like `Web3AuthSigner.fromSDKInstance(inner: Web3Auth)`